arelle/plugin/formulaLoader.py

```diff
@@ -1283,9 +1283,9 @@ def compileXfsGrammar( cntlr, debugParsing ):
     cntlr.showStatus(_("Compiling Formula Grammar"))
     from pyparsing import (Word, Keyword, alphas,
                  Literal, CaselessLiteral,
-                 Combine, Optional, nums, Or, Forward, Group, ZeroOrMore, OneOrMore, StringEnd, alphanums,
-                 ParserElement, quotedString, dblQuotedString, sglQuotedString, QuotedString,
-                 delimitedList, Suppress, Regex, FollowedBy,
+                 Combine, Opt, nums, Or, Forward, Group, ZeroOrMore, OneOrMore, StringEnd, alphanums,
```

Comment:
Whats the difference between `Optional` and `Opt` I saw you changed a bunch of them?

Reply:
They're the same. pyparsing deprecated `Optional` when a class of the same name was created for the base python `typing` module.